contribs/application/src/test/java/org/matsim/application/options/ShpOptionsTest.java

```diff
@@ -61,4 +63,35 @@ public void all() {
 				.hasSize(Set.copyOf(ft).size());
 
 	}
+
+	@Test
+	public void testGetGeometry() {
```

Comment:
I found it quite difficult to understand the intend of this test. Naming the variables geometry1,2,3 makes it very hard to understand what the purpose of those variables are. I think a better naming for `geometry1` would be `joinedResult` and for `geometry2` `expectedResult` for example. Of course you can choose any other name, which describes what these things are for. 

I think the comparison of `geometry1` and `geometry3` doesn't make sense. You want to test whether this method correctly joins all the geometries. This would be much clearer without `geometry3`. I can see why you have it in there but the test would also fail if only the geometry of the first feature was returned, as it was the case before your fix. 

Of course one test is better than no test! Nevertheless, I can see four execution paths in the `getGeometry` method. If you want to do it right you could add tests for:

- No features - Exception is expected
- One feature - This could be done the way you handled `geometry3`
- Multiple features - Some feature without geometry 
- Multiple features - Everything is as expected -> The happy path (your test)

Reply:
I did some renaming and deleted gemeotry3. For now I won't add further tests, as I have lots of other stuff to do.